metadata/images.go

```diff
@@ -177,7 +173,7 @@ func (s *imageStore) Update(ctx context.Context, image images.Image, fieldpaths
 			updated = image
 		}
 
-		if err := validateImage(&image); err != nil {
+		if err := validateImage(&updated); err != nil {
```

Comment:
Could you add a test case that fails without this change?

Reply:
I am trying to think of a test case here that would meet that criteria. A test could exist which errors with invalid values in `image` but are not in the field path and therefore do not get set in updated, however I am not sure that is a useful test. We currently don't enforce the overall label size so can't test that the update pushes the resulting `updated` into an invalid state. Every other invalid input I can think of would fail with both `image` and `updated`.

Reply:
@dmcgowan The problem here is that I've scoured this code twice and seen this mistake twice. I am not sure this actually matters if we don't think a test case can catch it.

A possible test case here would be to have a field that is validated with a wrong value, but not a part of the field path. Its possible none may exist for the image type.

Reply:
@yanxuean To be clear, no need to add a test for this case, as its not possible. ;)

Reply:
Yes, I agree